src/org/labkey/test/tests/AdminConsoleNavigationTest.java

```diff
@@ -71,7 +108,134 @@ public void testAdminNavTrails()
+
+        for (Map.Entry<String, String> link : linkHrefs.entrySet())
+        {
+            if (ignoredLinks.contains(link.getKey()))
+                TestLogger.log("Skipping admin link: " + link.getKey());
+            else
+            {
+                log("Verifying link " + link.getKey() + " with URL " + link.getValue());
+                verifyReadOnlyLinks(link.getKey(), link.getValue());
+                verifyPostCommandFails(link.getValue());
+            }
+        }
+        goToHome();
+        stopImpersonating();
+    }
+
+    @Test
+    public void testAdminConsoleLinksForAdminAndNonAdmin()
+    {
+        Set<String> ignoredLinksNonAdmin = Collections.newSetFromMap(new CaseInsensitiveHashMap<>());
+        ignoredLinksNonAdmin.addAll(List.of(
+                "Merge sync admin",     //Can be accessed by non admin
+                "Dump Heap",            // Undesired consequences
+                "Reset Site Errors",    // Undesired consequences
+                "Running Threads",      // Undesired consequences
+                "Credits"               //Can be accessed by non admin
+        ));
+
+        Set<String> ignoredLinksAdmin = Collections.newSetFromMap(new CaseInsensitiveHashMap<>());
+        ignoredLinksAdmin.addAll(List.of(
+                "Dump Heap",            // Undesired consequences
+                "Reset Site Errors",    // Undesired consequences
+                "Running Threads"      // Undesired consequences
+        ));
+        ShowAdminPage adminConsole = goToAdminConsole();
+        List<WebElement> adminLinks = adminConsole.getAllAdminConsoleLinks();
+        Map<String, String> linkHrefs = new LinkedHashMap<>();
+        for (WebElement link : adminLinks)
+            linkHrefs.put(link.getText(), link.getAttribute("href"));
+
+        log("Verifying links can be accessed by admin");
+        for (Map.Entry<String, String> link : linkHrefs.entrySet())
+        {
+            if (ignoredLinksAdmin.contains(link.getKey()))
+                TestLogger.log("Skipping admin link: " + link.getKey());
+            else
+            {
+                log("Verifying link " + link.getKey() + " with URL " + link.getValue());
+                verifyLinks(link.getKey(), link.getValue(), 200);
+            }
+        }
+
+        log("Verifying links cannot be accessed by non admin");
+        goToHome();
+        impersonate(NON_ADMIN);
+        for (Map.Entry<String, String> link : linkHrefs.entrySet())
+        {
+            if (ignoredLinksNonAdmin.contains(link.getKey()))
+                TestLogger.log("Skipping admin link: " + link.getKey());
+            else
+            {
+                log("Verifying link " + link.getKey() + " with URL " + link.getValue());
+                verifyLinks(link.getKey(), link.getValue(), 403);
+            }
+        }
+        goToHome();
+        stopImpersonating();
+    }
+
+    private void verifyReadOnlyLinks(String text, String url)
```

Comment:
This method only verifies a single link
```suggestion
    private void verifyReadOnlyLink(String text, String url)
```

Reply:
Done